core/src/main/scala/io/projectglow/vcf/VCFFileFormat.scala

```diff
@@ -68,12 +68,7 @@ class VCFFileFormat extends TextBasedFileFormat with DataSourceRegister with Hls
       )
     }
 
-    codecFactory.getCodec(path) match {
-      case null => true
-      case _: BGZFEnhancedGzipCodec =>
-        VCFFileFormat.isValidBGZ(path, sparkSession.sparkContext.hadoopConfiguration)
-      case c => c.isInstanceOf[SplittableCompressionCodec]
-    }
+    codecFactory.getCodec(path).isInstanceOf[SplittableCompressionCodec]
```

Comment:
Can you add a comment saying that we'll lazily determine whether this is bgzipped later?